internal/providers/terraform/azure/virtual_machine_scale_set.go

```diff
@@ -0,0 +1,84 @@
+package azure
+
+import (
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+	"github.com/tidwall/gjson"
+)
+
+func GetAzureRMVirtualMachineScaleSetRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "azurerm_virtual_machine_scale_set",
+		RFunc: NewAzureRMVirtualMachineScaleSet,
+	}
+}
+
+func NewAzureRMVirtualMachineScaleSet(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	costComponents := []*schema.CostComponent{}
+	subResources := []*schema.Resource{}
+
+	location := d.Get("location").String()
+	instanceType := d.Get("sku.0.name").String()
+	capacity := decimal.NewFromInt(d.Get("sku.0.capacity").Int())
+
+	if u != nil && u.Get("instances").Type != gjson.Null {
+		capacity = decimal.NewFromInt(u.Get("instances").Int())
+	}
+
+	os := "Linux"
+	if d.Get("os_profile_windows_config").Type != gjson.Null {
+		if d.Get("storage_profile_os_disk.0.os_type").String() == "Windows" {
+			os = "Windows"
+		}
+	}
```

Comment:
Looks like the `azurerm_virtual_machine_scale_set` resource has a `storage_profile_image_reference.0.offer` which can be set to "WindowsServer" as well, so I think we should also check that, like what we do with `storage_image_reference.0.offer` for the VM resource.